Userland/Libraries/LibGL/Tex/TextureUnit.cpp

```diff
@@ -11,29 +11,22 @@ namespace GL {
 
 void TextureUnit::bind_texture_to_target(GLenum texture_target, const RefPtr<Texture>& texture)
 {
-    switch (texture_target) {
-    case GL_TEXTURE_2D:
-        m_texture_target_2d = static_ptr_cast<Texture2D>(texture);
-        m_currently_bound_texture = texture;
-        m_currently_bound_target = GL_TEXTURE_2D;
-        break;
-    default:
-        VERIFY_NOT_REACHED();
+    if (!texture) {
+        m_texture_target_2d = nullptr;
+        m_currently_bound_target = GL_NONE;
+        m_currently_bound_texture = nullptr;
+        return;
     }
-}
 
-void TextureUnit::unbind_texture(GLenum texture_target)
-{
     switch (texture_target) {
     case GL_TEXTURE_2D:
-        m_texture_target_2d = nullptr;
-        m_currently_bound_target = 0;
+        m_texture_target_2d = static_ptr_cast<Texture2D>(texture);
+        m_currently_bound_target = GL_TEXTURE_2D;
+        m_currently_bound_texture = texture;
         break;
     default:
         VERIFY_NOT_REACHED();
     }
-
-    m_currently_bound_texture = nullptr;
```

Comment:
After consulting the spec again this seems wrong (it was already wrong before). Each texture unit can have all kinds of texture targets bound at the same time (1D, 2D, Cube, etc.) The texture target that is actually used by the rasterizer is determined by the enabled targets ( `glEnable(GL_TEXTURE_CUBE)` ) and the priority (Cube before 2D, 2D before 1D). So `m_currently_bound_target` and `m_currently_bound_texture` don't really make sense.

https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glBindTexture.xhtml

Up to you if you want to fix this now, I'd also accept these changes as is since they don't make the situation any worse. :^)

Reply:
I came to the same conclusion as you, but I've chosen not to implement the 1D/3D/cubemap texture targets as well and leave the code as-is for now, since it will be long before we start implementing those other targets.